netty/src/main/java/io/grpc/transport/netty/NettyClientHandler.java

```diff
@@ -241,15 +241,19 @@ public boolean visit(Http2Stream stream) throws Http2Exception {
   }
 
   @Override
+  public void exceptionCaught(ChannelHandlerContext ctx, Throwable cause) throws Exception {
```

Comment:
Not connection failure itself, but yes, I think TLS failures. But it would also handle more mundane things like `NullPointerException` or `IllegalStateException`.